### PR TITLE
Additional token patch fields

### PR DESCRIPTION
**Pull request by paulcjh**

### CODE REVIEW

**Review by isNeil — APPROVED** [review]
LGTM

File: pipeline/schemas/token.py [hunk 1]
```diff
@@ -40,6 +40,10 @@ class TokenPatch(Patchable):
     name: Optional[str]
     # Boolean specifying if token can be used
     is_enabled: Optional[bool]
+    # Token value (only sudo tokens can use this)
+    token: Optional[str]
+    # Datetime to expire the token
+    expire_at: Optional[datetime]
```
Comment (alexpearce) [inline pipeline/schemas/token.py:46]: @plutopulp and I had a chat about this and concluded that letting this field be patched defeats the purpose of having it. If a token has an expiry date but gets leaked the attacker can just set the expiry date to far in the future using the leaked token.

GitHub provides a feature to create a new token with an identical feature set but with a new expiry and value. Going down that route feels nicer to me.

File: pipeline/schemas/token.py [hunk 2]
```diff
@@ -40,6 +40,10 @@ class TokenPatch(Patchable):
     name: Optional[str]
     # Boolean specifying if token can be used
     is_enabled: Optional[bool]
+    # Token value (only sudo tokens can use this)
+    token: Optional[str]
```
Comment (alexpearce) [inline pipeline/schemas/token.py:44]: Do you have a corresponding pipeline-top PR for this? What's the use case here?

Comment (paulcjh) [reply pipeline/schemas/token.py:44]: APIv1 migration

Comment (paulcjh) [reply pipeline/schemas/token.py:44]: I need to be able to set user's tokens to what they were in APIv1

File: pipeline/schemas/token.py (hunk 1 above)
Comment (paulcjh) [reply pipeline/schemas/token.py:46]: Do all tokens have the same permissions for users?
A reasonable use case here would be if someone needs to extend a production token expiry date.

Comment (paulcjh) [reply pipeline/schemas/token.py:46]: Also it's possible that the expire_at field can only be set by dashboard token

**Pull request merged**